pkg/apis/config/registration.go

```diff
@@ -14,12 +14,29 @@ limitations under the License.
 
 package config
 
+import (
+	"knative.dev/pkg/configmap"
+)
+
 // Constructor should take the form func(*v1.ConfigMap) (T, error)
 type Constructor interface{}
 
 // Registration defines a ConfigMap registration to be watched by the settingsstore.Watcher
 // and to be injected into the Reconcile() contexts of controllers
 type Registration struct {
 	ConfigMapName string
-	Constructor   Constructor
+	Constructor   interface{}
+	DefaultData   map[string]string
+}
+
+func (r Registration) ValidateOrDie() {
```

Comment:
I'd lean towards APIs having a Validate method that returns an error (like our other APIs do) and delegating to the caller for what to do about it.